libraries/chain/controller.cpp

```diff
@@ -4128,8 +4124,11 @@ struct controller_impl {
       log_and_drop_future<void> verify_qc_future;
       if constexpr (is_proper_savanna_block) {
          if (qc) {
-            verify_qc_future = post_async_task(thread_pool.get_executor(), [this, &qc, &prev] {
-               verify_qc(prev, *qc);
+            // do basic checks on provided qc immediately (excluding signature verification)
+            prev.verify_qc_basic(*qc);
```

Comment:
Seems like this should be done inside the posted task below. It is unlikely to ever fail. Accessing the same data on the same thread would be better than accessing it on two different threads.

Reply:
https://github.com/AntelopeIO/spring/pull/927/commits/d4b6861248443223d09a273ace17fe5db9d633f3